apps/accounts/models/provider_request.py

```diff
@@ -403,3 +446,29 @@ def clean(self) -> None:
             raise ValidationError(f"{reason}, you haven't submitted either.")
         if self.link and bool(self.file):
             raise ValidationError(f"{reason}, you've attempted to submit both.")
+
+    def has_content_match(self, other_doc: "HostingProviderSupportingDocument") -> bool:
```

Comment:
This is where we push much of the logic for comparing submitted evidence to already uploaded supporting documents.
This currently only supports checking evidence against supporting docs - not the other way around, and not evidence against evidence, or supporting doc against other supporting docs.